aten/src/ATen/core/List_inl.h

```diff
@@ -129,6 +129,12 @@ template<class T, class Iterator>
 void swap(ListElementReference<T, Iterator>&& lhs, ListElementReference<T, Iterator>&& rhs) {
   std::swap(*lhs.iterator_, *rhs.iterator_);
 }
+
+template<class T, class Iterator>
+bool operator==(const ListElementReference<T, Iterator>& lhs, const T& rhs) {
+  T lhs_tmp = lhs;
```

Comment:
this overload looks fine to me. cc @smessmer will this custom == overload have any implications on the list impl?

Reply:
Let's restrict it to only work on `const ListElementReference<T, Iterator>& rhs`. `operator==` should be symmetric or the API gets weird because `a == b` does something different than `b == a`.

Reply:
@smessmer just to confirm, is the suggestion to add a mirror version of this overload as well, or just not have overload at all? 

If without this overload, then std::find() will not compile (see https://github.com/pytorch/pytorch/pull/40327#discussion_r443090950)

Reply:
> @smessmer just to confirm, is the suggestion to add a mirror version of this overload as well, or just not have overload at all?
> 
> If without this overload, then std::find() will not compile (see [#40327 (comment)](https://github.com/pytorch/pytorch/pull/40327#discussion_r443090950))

@smessmer ping :)

Reply:
ok sg. I meant to remove the overload, but `std::find` convinced me. Having it symmetrical is fine.